src/System.ServiceModel.UnixDomainSocket/src/System/ServiceModel/UnixDomainSocketTransportSecurity.cs

```diff
@@ -0,0 +1,182 @@
+            {
+                throw DiagnosticUtility.ExceptionUtility.ThrowHelperError(new InvalidOperationException(SR.Format(
+                    SR.UnsupportedSslProtectionLevel, _protectionLevel)));
+            }
+
+            SslStreamSecurityBindingElement result = new SslStreamSecurityBindingElement();
+            result.RequireClientCertificate = requireClientCertificate;
+            result.SslProtocols = _sslProtocols;
+            return result;
+        }
+
+        internal BindingElement CreatePosixIdentityOnlyBinding()
+        {
+            return new UnixPosixIdentityBindingElement();
+        }
+
+        private static bool IsSslBindingElement(BindingElement element, UnixDomainSocketTransportSecurity transportSecurity)
+        {
+            SslStreamSecurityBindingElement ssl = element as SslStreamSecurityBindingElement;
+            if (ssl == null)
+            {
+                return false;
+            }
+
+            transportSecurity.ProtectionLevel = ProtectionLevel.EncryptAndSign;
+            return true;
+        }
+
+        internal BindingElement CreateTransportProtectionOnly()
+        {
+            return CreateSslBindingElement(false);
+        }
+
+        internal static bool SetTransportProtectionOnly(BindingElement transport, UnixDomainSocketTransportSecurity transportSecurity)
+        {
+            return IsSslBindingElement(transport, transportSecurity);
+        }
+
+        internal BindingElement CreateTransportProtectionAndAuthentication()
+        {
+            if (_clientCredentialType == UnixDomainSocketClientCredentialType.Default)
+            {
+                if (RuntimeInformation.IsOSPlatform(OSPlatform.Windows))
+                {
+                    return new WindowsStreamSecurityBindingElement
+                    {
+                        ProtectionLevel = _protectionLevel
+                    };
+                }
+                else
+                {
+                    return CreatePosixIdentityOnlyBinding();
+                }
+            }else if (_clientCredentialType == UnixDomainSocketClientCredentialType.Certificate )
+            {
+                return CreateSslBindingElement(true);
+
+            }else if(_clientCredentialType == UnixDomainSocketClientCredentialType.Windows)
+            {
+                return new WindowsStreamSecurityBindingElement
+                {
+                    ProtectionLevel = _protectionLevel
+                };
+
+            }else if (_clientCredentialType == UnixDomainSocketClientCredentialType.IdentityOnly)
+            {
+                if (RuntimeInformation.IsOSPlatform(OSPlatform.Windows))
+                {
+                    throw new NotSupportedException();
+                }
+                return CreatePosixIdentityOnlyBinding();
+            }
+            else
+            {
+                return null;
+            }
+        }
+
+        internal bool InternalShouldSerialize()
+        {
```

Comment:
Remove, this was used when writing out the binding to config and would be used to know if it was using defaults or not. If using defaults, no need to write it out to the xml. This is left over dead code.